kong/plugins/ip-restriction/handler.lua

```diff
@@ -2,10 +2,48 @@ local BasePlugin = require "kong.plugins.base_plugin"
 local responses = require "kong.tools.responses"
 local iputils = require "resty.iputils"
 
+
+local ok, new_tab = pcall(require, "table.new")
+if not ok then
+  new_tab = function() return {} end
+end
+
+
+-- cache of parsed CIDR values
+local cache = {}
+
+
 local IpRestrictionHandler = BasePlugin:extend()
 
 IpRestrictionHandler.PRIORITY = 990
 
+local function cidr_cache(cidr_tab)
+  local parsed_cidrs = new_tab(#cidr_tab, 0) -- table of parsed cidrs to return
```

Comment:
It seems we are trying to ensure performance in a hot code path here. As such, it would make sense to cache the result of `#cidr_tab`, as it is being called twice in this function.